### ISSUE TITLE
TransferService (S3) crashing for Pixel 3a (Android 10) on background

### ISSUE THREAD

**Issue by arindamxd** [aws-amplify/aws-sdk-android#1193]
**Describe the bug**
Trying to upload data to S3 while in the background it's crashing on Android 10 devices.

Log
```
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err: java.lang.IllegalStateException: Not allowed to start service Intent { act=add_transfer cmp=com.example/com.amazonaws.mobileconnectors.s3.transferutility.TransferService (has extras) }: app is in background uid UidRecord{8c313fc u0a226 TRNB idle change:uncached procs:1 seq(0,0,0)}
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1616)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at android.app.ContextImpl.startService(ContextImpl.java:1571)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at android.content.ContextWrapper.startService(ContextWrapper.java:669)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at com.amazonaws.mobileconnectors.s3.transferutility.TransferUtility.sendIntent(TransferUtility.java:432)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at com.amazonaws.mobileconnectors.s3.transferutility.TransferUtility.upload(TransferUtility.java:216)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at com.amazonaws.mobileconnectors.s3.transferutility.TransferUtility.upload(TransferUtility.java:186)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at com.amazonaws.mobileconnectors.s3.transferutility.TransferUtility.upload(TransferUtility.java:154)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at com.example.worker.UploadWorker.a(Unknown Source:61)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at com.example.worker.UploadWorker.onTokenFetchSuccess(Unknown Source:97)
2019-09-12 19:08:09.289 7328-7328/com.example W/System.err:     at com.example.worker.UploadWorker.startWork(Unknown Source:158)
```

**Which AWS service(s) are affected?**
S3 & TransferService

**Expected behavior**
Using WorkManager instead of using Service/IntentService may solve the problem.

**Environment Information (please complete the following information):**
 - AWS Android SDK Version: 2.2.+
 - Device: Pixel 3a, Simulator
 - Android Version: Android 10

**Comment by desokroshan** [comment]
Calling `startService` when the app is in the background will crash your application on devices running Oreo and above. This is part of limitations introduced by Android on background processing for devices running Oreo and above : https://developer.android.com/about/versions/oreo/background. 

The solution for devices running Oreo and above would be to invoke `startForegroundService` with a valid notification object passed through the intent to `startForegroundService`.

Please refer to [Supporting TransferService on Oreo and above](https://aws-amplify.github.io/docs/android/storage#supporting-transferservice-on-oreo-and-above) for additional details. Hope that helps!

**Comment by arindamxd** [comment]
Thanks, fixed it by upgrading Core + S3 version to **2.15.+**.
